api/tests/unit/integrations/lead_tracking/hubspot/test_unit_hubspot_client.py

```diff
@@ -44,10 +50,34 @@ def test_create_lead_form(
     )
 
     # When
-    response = hubspot_client.create_lead_form(staff_user, hubspot_cookie)
+    response = hubspot_client.create_lead_form(staff_user, hubspot_cookie_body)
 
     # Then
+    assert len(responses.calls) == 1
     assert response == {"inlineMessage": "Thanks for submitting the form."}
+    call: responses.Call = responses.calls[0]  # type: ignore[assignment]
+    request_body = json.loads(call.request.body)
+    assert response == {"inlineMessage": "Thanks for submitting the form."}
+
+    fields: list[dict[str, str]] = request_body["fields"]
+
+    assert {"objectTypeId": "0-1", "name": "email", "value": staff_user.email} in fields
+    assert {
+        "objectTypeId": "0-1",
+        "name": "firstname",
+        "value": staff_user.first_name,
+    } in fields
+    assert {
+        "objectTypeId": "0-1",
+        "name": "lastname",
+        "value": staff_user.last_name,
+    } in fields
+
+    context = request_body.get("context", {})
+    if hubspot_cookie_body is not None:
+        assert context["hutk"] == hubspot_cookie_body
+    else:
+        assert context == {}
```

Comment:
I'm not a big fan of having conditional assert statements in tests - I'd rather create these as separate tests if possible, but I appreciate that might result in a fair amount of code duplication - is there a way we can share the code without having these conditional assert statements?

Reply:
True, better with parametrization thanks